### ISSUE TITLE
Add support for Amazon Linux - AL2023 (x86_64 and aarch64) 

### ISSUE THREAD

**Issue by danie-dejager** [openzfs/zfs#15221]
### Describe the feature would like to see added to OpenZFS

Please consider adding supported packages for Amazon Linux 2023 (AL2023)
https://aws.amazon.com/linux/amazon-linux-2023/

AL2023 is a distribution based on the combination of packages from various Fedora releases (34..37). As time moves on I will no longer be able to use Fedora packages to install ZFS on my AL2023 hosts as the Fedora version itself will no longer be supported by Fedora. AL2023 will be supported until 2028.

### How will this feature improve OpenZFS?

I will be able to install OpenZFS on my AL2023 hosts (x86_64 and aarch64) without running the risk of using a package that was not 100% intended for the platform it is installed on.

Amazon can't package OpenZFS as can be seen from the request I made earlier the year:
https://github.com/amazonlinux/amazon-linux-2023/issues/262
My intent was for them to repackage the OpenZFS packges and not make OpenZFS a part of their kernel though.

**Comment by stewartsmith** [comment]
Note that you can build AL2023 packages on COPR (https://copr.fedorainfracloud.org/) so this may "just work" if existing RPM packaging is submitted to build.

**Comment by sklarsa** [comment]
> Note that you can build AL2023 packages on COPR (https://copr.fedorainfracloud.org/) so this may "just work" if existing RPM packaging is submitted to build.

Unfortunately, there's another issue with AL2023 and COPR that's preventing this solution from working out-of-the-box :-( https://github.com/amazonlinux/amazon-linux-2023/issues/643

**Comment by allentc** [comment]
Apparently [https://github.com/amazonlinux/amazon-linux-2023/issues/643](https://github.com/amazonlinux/amazon-linux-2023/issues/643) was solved a few months ago.  Is this sufficient?

**Comment by SpComb** [comment]
Trying to build [custom packages](https://openzfs.github.io/openzfs-docs/Developer%20Resources/Custom%20Packages.html) from the latest `zfs-2.4-release` branch on amzn2023 with `kernel6.18`, the `zfs-dkms` package is not installable due to a conflict with the higher epoch on the amzn2023 kernel packages:

```
$ sudo dnf install zfs-dkms-2.4.3-1.amzn2023.noarch.rpm
Last metadata expiration check: 3:29:29 ago on Mon Jul 13 11:00:08 2026.
Error:
 Problem: problem with installed package kernel6.18-devel-1:6.18.36-69.138.amzn2023.x86_64
  - package zfs-dkms-2.4.3-1.amzn2023.noarch from @commandline conflicts with kernel-devel > 7.0.999 provided by kernel6.18-devel-1:6.18.36-69.138.amzn2023.x86_64 from @System
  - package zfs-dkms-2.4.3-1.amzn2023.noarch from @commandline conflicts with kernel-devel > 7.0.999 provided by kernel6.18-devel-1:6.18.36-69.138.amzn2023.x86_64 from amazonlinux
  - conflicting requests
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages)
```

Tested to build and install with the following patch to depend on `kernel-devel-uname-r` without the epoch instead:

```patch
diff --git a/rpm/generic/zfs-dkms.spec.in b/rpm/generic/zfs-dkms.spec.in
index fe127025e..167fb153e 100644
--- a/rpm/generic/zfs-dkms.spec.in
+++ b/rpm/generic/zfs-dkms.spec.in
@@ -31,7 +31,16 @@ Requires:       gcc, make, perl, diffutils
 Requires(post): gcc, make, perl, diffutils
 
 # Hold back kernel upgrades if kernel is not supported by ZFS
-%if 0%{?rhel}%{?fedora}%{?mageia}%{?suse_version}%{?openEuler}
+%if 0%{?amzn} == 2023
+Requires:       kernel-devel-uname-r >= @ZFS_META_KVER_MIN@, kernel-devel-uname-r <= @ZFS_META_KVER_MAX@.999
+Requires(post): kernel-devel-uname-r >= @ZFS_META_KVER_MIN@, kernel-devel-uname-r <= @ZFS_META_KVER_MAX@.999
+Conflicts:      kernel-devel-uname-r < @ZFS_META_KVER_MIN@, kernel-devel-uname-r > @ZFS_META_KVER_MAX@.999
+Requires:       kernel-uname-r >= @ZFS_META_KVER_MIN@, kernel-uname-r <= @ZFS_META_KVER_MAX@.999
+Requires(post): kernel-uname-r >= @ZFS_META_KVER_MIN@, kernel-uname-r <= @ZFS_META_KVER_MAX@.999
+Conflicts:      kernel-uname-r < @ZFS_META_KVER_MIN@, kernel-uname-r > @ZFS_META_KVER_MAX@.999
+
+Obsoletes:      spl-dkms <= %{version}
+%elif 0%{?rhel}%{?fedora}%{?mageia}%{?suse_version}%{?openEuler}
 Requires:       kernel-devel >= @ZFS_META_KVER_MIN@, kernel-devel <= @ZFS_META_KVER_MAX@.999
 Requires(post): kernel-devel >= @ZFS_META_KVER_MIN@, kernel-devel <= @ZFS_META_KVER_MAX@.999
 Conflicts:      kernel-devel < @ZFS_META_KVER_MIN@, kernel-devel > @ZFS_META_KVER_MAX@.999
```
